dev/sensor/bh1750/export/dev/sensor/bh1750.hpp

```diff
@@ -0,0 +1,239 @@
+    low,
+    high,
+};
+
+//! \brief Represents resolution mode BH1750 sensor.
+//!  According to RM, there are 3 resolution modes:
+//!  L-mode (4lx), H-mode (1lx) and H-mode2 (0.5lx)
+//!
+enum class resolution {
+    low,
+    high,
+    high2
+};
+
+}; // bh1750_cfg
+
+
+//! \brief BH1750 Ambient Light sensor driver implementation.
+//! \tparam i2c_dev I2C generic bus driver
+//! \tparam address I2C slave address.
+//!  This parameter can be value of type bh1750_cfg::i2c_address
+//!
+template <class i2c_dev, bh1750_cfg::i2c_address address>
+class bh1750 {
+public:
+    bh1750() = delete;
+    ~bh1750() = delete;
+
+    //! \brief Inits sensor and underlying I2C platform bus.
+    //! \retval Status of operation.
+    //!
+    static ecl::err init();
+
+    //! \brief Sets sensor resolution mode. Default mode is bh1750_cfg::high.
+    //! \param[in] resolution Resolution mode.
+    //!  This parameter can be value of type bh1750_cfg::resolution.
+    //!
+    static void set_resolution(bh1750_cfg::resolution resolution);
+
+    //! \brief Returns current sensor resolution mode.
+    //! \retval Current resolution mode.
+    //!  The return value can be value of type bh1750_cfg::resolution.
+    //!
+    static bh1750_cfg::resolution get_resolution();
+
+    //! \brief Reads raw illuminance sample from sensor.
+    //! \details This method blocks until measurement is not finished.
+    //!  Measurement time depends on resolution mode: at least 25 ms for low mode
+    //!  and 200 ms for high and high2 modes.
+    //! \param[out] sample  Raw sample value will be written in this parameter.
+    //! \retval Status of the operation.
+    //!
+    static ecl::err get_illuminance_sample(uint16_t &sample);
+
+    //! \brief Reads sample from sensor and converts to a physical value.
+    //! \details This method blocks until measurement is not finished.
+    //!  Measurement time depends on resolution mode: at least 25 ms for low mode
+    //!  and 200 ms for high and high2 modes.
+    //! \param[out] value Illuminance in (1000 * (illuminance in lx))
+    //!  will be written in this parameter.
+    //! \retval Status of the operation.
+    //!
+    static ecl::err get_illuminance(uint32_t &value);
+
+private:
+    //! Returns I2C slave address based on class template parameter
+    static constexpr uint8_t pick_i2c_slave_address();
+
+    //! I2C slave address when ADDR pin is pulled down to GND
+    static constexpr uint8_t i2s_address_low = 0x46;
+    //! I2C slave address when ADDR pin is pulled up to VCC
+    static constexpr uint8_t i2s_address_high = 0xB8;
+
+    //! Maximum measurement time for low resolution mode according to RM
+    static constexpr uint16_t measurement_time_ms_low = 25;
+    //! Maximum measurement time for high and high2 resolution modes according to RM
+    static constexpr uint16_t measurement_time_ms_high = 200;
+
+    //! Current resolution mode
+    static bh1750_cfg::resolution resolution;
```

Comment:
Could you please prefix it with `m_` part, that's how class members are distinguished from other variables or names.
It very useful when using auto-complete and for redability.